controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -293,6 +289,18 @@ func recordActionFailure(info *reconcileInfo, errorType metal3v1alpha1.ErrorType
 	return actionFailed{dirty: true, ErrorType: errorType, errorCount: info.host.Status.ErrorCount}
 }
 
+func recordActionFailureForBusyProvisioner(info *reconcileInfo) actionFailed {
```

Comment:
I'm thinking that we will need to move the state machine file to its own module soon, so it would be better if this were defined in that file and didn't have any dependencies on stuff in baremetalhost_controller.go.